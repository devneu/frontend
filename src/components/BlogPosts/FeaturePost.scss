.post-feature-card {
  border-radius: 5px;
  margin: 10px 0px;

  img {
    border-radius: 5px 5px 0px 0px;
  }

  .button-feature-card {
    background-color: #1890ff;
    color: white;
  }
}