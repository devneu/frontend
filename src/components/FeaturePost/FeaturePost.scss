.post-feature-card {
  border-radius: 5px;

  img {
    border-radius: 5px 5px 0px 0px;
  }

  .button-feature-card {
    background-color: #1890ff;
    color: white;
  }
}