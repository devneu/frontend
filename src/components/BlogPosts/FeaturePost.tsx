import { Card, Typography } from 'antd';
import React from 'react';
import './FeaturePost.scss';
import { Link } from 'react-router-dom';
import { Post } from '../../types/Post.Interface';

const { Title, Paragraph } = Typography;

interface FeaturePostProps {
  post: Post;
}

const FeaturePost = ({ post }: FeaturePostProps) => (
  <Card className="post-feature-card" cover={<img alt="card-img" src={post.postImg} />}>
    <Card.Meta description={post.postDate} />
    <Title level={2}>{post.postTitle}</Title>
    <Paragraph className="post-text">{post.postText}</Paragraph>
    <Link to={`/blog/${post.id}`}>Read more</Link>
  </Card>
);

export default FeaturePost;
