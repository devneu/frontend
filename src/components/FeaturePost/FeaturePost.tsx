import { Button, Card, Typography } from 'antd';
import React from 'react';
import { ArrowRightOutlined } from '@ant-design/icons';
import { NavLink } from 'react-router-dom';
import { Post } from '../../types/Post.interface';
import './FeaturePost.scss';

interface FeaturePostProps {
  post: Post;
}

const FeaturePost = ({ post }: FeaturePostProps) => (
  <Card className="post-feature-card" cover={<img alt="card-img" src={post.img} />}>
    <Card.Meta description={post.date} />
    <Typography.Title level={2}>{post.title}</Typography.Title>
    <Typography.Paragraph className="post-text">{post.text}</Typography.Paragraph>
    <NavLink to={`/blog/${post.id}`}>
      <Button type="primary">
        Read more
        <ArrowRightOutlined />
      </Button>
    </NavLink>
  </Card>
);

export default FeaturePost;
