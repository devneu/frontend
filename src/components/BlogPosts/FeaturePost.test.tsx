import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import FeaturePost from './FeaturePost';

describe('FeaturePost', () => {
  it('Should successfully render', () => {
    const wrapper = renderer
      .create(
        <BrowserRouter>
          <FeaturePost
            post={{
              id: 1,
              postImg: 'post image',
              postDate: ' post date',
              postTitle: ' post title',
              postText: 'post text',
            }}
          />
        </BrowserRouter>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
