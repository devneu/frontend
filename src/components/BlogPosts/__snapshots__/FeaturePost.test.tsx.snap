// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturePost Should successfully render 1`] = `
<div
  className="ant-card ant-card-bordered post-feature-card"
>
  <div
    className="ant-card-cover"
  >
    <img
      alt="card-img"
      src="post image"
    />
  </div>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-card-meta"
    >
      <div
        className="ant-card-meta-detail"
      >
        <div
          className="ant-card-meta-description"
        >
           post date
        </div>
      </div>
    </div>
    <h2
      className="ant-typography"
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
       post title
    </h2>
    <div
      className="ant-typography post-text"
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      post text
    </div>
    <a
      href="/blog/1"
      onClick={[Function]}
    >
      Read more
    </a>
  </div>
</div>
`;
